dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaYarnApplicationsTask.java

```diff
@@ -0,0 +1,177 @@
+            String.format(
+                "Dump YARN applications with %s type from %s cluster", yarnAppType, clusterName));
+        List<ApiYARNApplicationDTO> yarnApplications = appLoader.load(clusterName, yarnAppType);
+        totalYARNApplications.addAll(yarnApplications);
+        LOG.info(
+            String.format(
+                "Dumped %d YARN applications with %s type from %s cluster",
+                yarnApplications.size(), yarnAppType, clusterName));
+      }
+    }
+
+    String yarnAppsJson = parseObjectToJsonString(totalYARNApplications);
+    try (Writer writer = sink.asCharSink(StandardCharsets.UTF_8).openBufferedStream()) {
+      writer.write(yarnAppsJson);
+    }
+  }
+
+  private String buildISODateTime(int deltaInDays) {
+    ZonedDateTime dateTime =
+        ZonedDateTime.of(LocalDateTime.now().minusDays(deltaInDays), ZoneId.of("UTC"));
+    return dateTime.format(isoDateTimeFormatter);
+  }
+
+  private List<String> fetchYARNApplicationTypes(ClouderaManagerHandle handle, String clusterName) {
+    List<String> yarnApplicationTypes = new ArrayList<>(predefinedAppTypes);
+    String yarnAppTypesUrl =
+        handle.getApiURI().toString() + "clusters/" + clusterName + "/serviceTypes";
+    CloseableHttpClient httpClient = handle.getHttpClient();
+    JsonNode yarnAppTypesJson;
+    ApiYARNApplicationTypeDTO yarnAppTypesFromClouderaAPI;
+    try (CloseableHttpResponse chart = httpClient.execute(new HttpGet(yarnAppTypesUrl))) {
+      yarnAppTypesJson = readJsonTree(chart.getEntity().getContent());
+      yarnAppTypesFromClouderaAPI =
+          parseJsonStringToObject(yarnAppTypesJson.toString(), ApiYARNApplicationTypeDTO.class);
+      yarnApplicationTypes.addAll(yarnAppTypesFromClouderaAPI.getItems());
+    } catch (IOException ex) {
+      throw new RuntimeException(ex.getMessage(), ex);
+    }
+    return yarnApplicationTypes;
+  }
+
+  private class YarnApplicationsLoader {
+    private final String host;
+    private final CloseableHttpClient httpClient;
+
+    public YarnApplicationsLoader(String host, CloseableHttpClient httpClient) {
+      this.host = host;
+      this.httpClient = httpClient;
+    }
+
+    public List<ApiYARNApplicationDTO> load(String clusterName, String appType) {
+      List<ApiYARNApplicationDTO> yarnApplications = new LinkedList<>();
+      final int limit = 100;
+      int offset = 0;
+      boolean nextLoad = true;
+
+      while (nextLoad) {
+        List<ApiYARNApplicationDTO> newLoad = load(clusterName, appType, limit, offset);
+        yarnApplications.addAll(newLoad);
+        nextLoad = (!newLoad.isEmpty());
+        offset += limit;
+      }
+
+      return yarnApplications;
+    }
+
+    private List<ApiYARNApplicationDTO> load(
+        String clusterName, String appType, int limit, int offset) {
+      String yarnApplicationsUrl =
+          host + "clusters/" + clusterName + "/services/yarn/yarnApplications";
+      String fromDate = buildISODateTime(includedLastDays);
```

Comment:
This variable can be build only once for all class.

Reply:
Fixed.